app/services/cache/MongoCacheClient.scala

```diff
@@ -239,8 +274,14 @@ class MongoCacheClient(appConfig: ApplicationConfig, db: () => DefaultDB, applic
         acc + (value._1 -> Json.parse(plainText.value))
       }
     })
-
-    collection.update(ordered = false).one(bsonIdQuery(cache.id), BSONDocument("$set" -> Json.toJson(rebuiltCache)), upsert = true) map handleWriteResult
+    collection.findOneAndUpdate(
+      filter= bsonIdQuery(cache.id),
+      update = Updates.combine(
+        Updates.set("id", rebuiltCache.id),
+        Updates.set("data",Codecs.toBson(rebuiltCache.data)),
+        Updates.set("lastUpdated", LocalDateTime.now(ZoneOffset.UTC))),
+      options = FindOneAndUpdateOptions().upsert(true).returnDocument(ReturnDocument.AFTER)
+    ).toFuture.map(result => true)
```

Comment:
❓ Should we replace this `true` with unit if it always succeeds?

Reply:
will look into this

Reply:
This is a little more awkward as code within the controller depends on it being a bool - I think it may be to do with errors and such.

Reply:
Exactly. So my suggestion was to modify the controller code and only leave the "happy" path - assume this function returns true, delete - now dead - `else` branches.